public/app/features/variables/state/actions.ts

```diff
@@ -582,3 +619,33 @@ export const cancelVariables = (
   dependencies.getBackendSrv().cancelAllInFlightRequests();
   dispatch(cleanUpVariables());
 };
+
+export const updateOptions = (
+  identifier: VariableIdentifier,
+  awaitUpdateOptions = true
+): ThunkResult<Promise<void>> => async (dispatch, getState) => {
+  const variableInState = getVariable(identifier.id, getState());
+
+  if (!awaitUpdateOptions) {
+    dispatch(variableStateFetching(toVariablePayload(variableInState)));
+    return variableAdapters
+      .get(variableInState.type)
+      .updateOptions(variableInState)
+      .then(() => {
+        dispatch(variableStateCompleted(toVariablePayload(variableInState)));
+      })
+      .catch(error => {
+        dispatch(variableStateFailed(toVariablePayload(variableInState, { error })));
+      });
+  }
+
+  try {
+    dispatch(variableStateFetching(toVariablePayload(variableInState)));
+    await variableAdapters.get(variableInState.type).updateOptions(variableInState);
+    dispatch(variableStateCompleted(toVariablePayload(variableInState)));
+  } catch (error) {
+    console.error(error);
+    dispatch(variableStateFailed(toVariablePayload(variableInState, { error })));
+    dispatch(notifyApp(createErrorNotification('Templating', 'Error updating options: ' + error.message)));
```

Comment:
These two path !awaitUpdateOptions  & else looks very similar, one is done with await the other is not, can you explain the difference?

Reply:
Yeah, it's ugly. In the first case we don't want to resolve variableAdapters.get(...).updateOptions but in the other we want to. Any suggestions on how to solve this. Could be two separate methods I guess?

Reply:
not sure I see the difference, one path is returning a promise of variableAdapters.get(...).updateOption,  the other is using await. keyword, both should return a similar promise with the same effect

Reply:
Hmm not sure, the first case is used in a Promise.all so I figured we wanted to run all Promises in that Promise.all in parallel. If I was to use await in within the Promise.all one wouldn't the Promise.all become sequential?

Reply:
Hmm, I think I'm wrong in my thinking here